File: app/src/main/java/com/conrradocamacho/desafio/network/bean/Movie.kt
package com.conrradocamacho.desafio.network.bean

import com.google.gson.annotations.SerializedName
import java.io.Serializable

/**
 * Created by Conrrado Camacho on 03/09/2019.
 * con.webmaster@gmail.com
 */
class Movie (
    val popularity: Double,

    @SerializedName("vote_count")
    val voteCount: Int,

    val video:Boolean,

    @SerializedName("poster_path")
    val posterPath: String?,

    val id: Long,
    val adult: Boolean,

    @SerializedName("backdrop_path")
    val backdropPath: String?,

    @SerializedName("original_language")
    val originalLanguage: String?,

    @SerializedName("original_title")
    val originalTitle: String?,

    @SerializedName("genre_ids")
    val genreIds: IntArray,

    val title: String?,

    @SerializedName("vote_average")
    val voteAverage: Double,

    val overview: String?,

    @SerializedName("release_date")
    val releaseDate: String?
): Serializable

File: MovieMaster/MovieMaster/Source/Models/Movie.swift
//
//  Movie.swift
//  MovieMaster
//
//  Created by Andre & Bianca on 30/11/19.
//  Copyright © 2019 Andre. All rights reserved.
//

import Foundation

struct MovieResult : Decodable {
    var page : Int?
    var totalResults : Int?
    var totalPages : Int?
    var results : [Movie]
}

struct Movie: Decodable {
    let id : Int?
    var title : String?
    let popularity : Double?
    let overview : String?

    let voteCount : Int?
    let voteAverage : Double?
    let posterPath : String?
    let originalLanguage : String?
    let originalTitle : String?
    let genreIds : [Int]?
    let backdropPath : String?
    let releaseDate : String?
}
